arcgis-ios-sdk-samples/Augmented reality/Navigate in AR/NavigateARRoutePlannerViewController.swift

```diff
@@ -0,0 +1,225 @@
+            self.setStatus(message: "Tap camera to start navigation.")
+        }
+    }
+    
+    // MARK: UI
+    
+    func setStatus(message: String) {
+        statusLabel.text = message
+    }
+    
+    // MARK: - Actions
+    
+    @IBAction func reset() {
+        routeGraphicsOverlay.graphics.removeAllObjects()
+        stopGraphicsOverlay.graphics.removeAllObjects()
+        routeParameters.clearStops()
+        routeResult = nil
+        startPoint = nil
+        endPoint = nil
+        resetButtonItem.isEnabled = false
+        navigateButtonItem.isEnabled = false
+        setStatus(message: "Tap to place a start point.")
+    }
+    
+    // MARK: UIViewController
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Add the source code button item to the right of navigation bar.
+        (self.navigationItem.rightBarButtonItem as? SourceCodeBarButtonItem)?.filenames = [
+            "NavigateARRoutePlannerViewController",
+            "NavigateARNavigatorViewController"
+        ]
+        // Avoid overlapping status label and map content.
+        mapView.contentInset.top = 2 * statusLabel.font.lineHeight
+        
+        // Configure the authentication manager to show the OAuth dialog.
+        AGSAuthenticationManager.shared().delegate = self
+        AGSAuthenticationManager.shared().oAuthConfigurations.add(oAuthConfiguration)
+        
+        routeTask.defaultRouteParameters { [weak self] (params: AGSRouteParameters?, error: Error?) in
+            guard let self = self else { return }
+            if let error = error {
+                self.presentAlert(error: error)
+                self.setStatus(message: "Loading route parameters failed.")
+                return
+            } else if let params = params {
+                // set the travel mode to the first one matching 'walking'
+                let walkMode = self.routeTask.routeTaskInfo().travelModes.first { $0.name.contains("Walking") }
+                params.travelMode = walkMode
+                params.returnStops = true
+                params.returnDirections = true
+                params.returnRoutes = true
+                self.routeParameters = params
+                self.mapView.touchDelegate = self
+                self.setStatus(message: "Tap to place a start point.")
+            }
+        }
+        
+        mapView.locationDisplay.start()
+    }
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if segue.identifier == "showNavigator" {
+            if let navigatorVC = segue.destination as? NavigateARNavigatorViewController {
+                navigatorVC.route = routeResult!
+                navigatorVC.routeTask = routeTask
+                navigatorVC.routeParameters = routeParameters
+            }
+        }
+    }
+    
+    deinit {
+        AGSAuthenticationManager.shared().oAuthConfigurations.removeAllObjects()
+    }
```

Comment:
It seems this line doesn't take effect - I can still access the auth-required API or resources after exiting the sample.

Reply:
Can use `AGSAuthenticationManager.shared().credentialCache.removeAllCredentials()` to clear the credentials. Don't know if it should be cleared.